### PR TITLE
feat : Refactor with hooks

### PR DESCRIPTION
**Pull request by taher-software**
In this branch , I've refactored the calculator components that I created in the last projects. Thus you will find the following changes :100: 
- Refactor the Calculator component (components/Calculator.js,ligneOfButton.js,resultIcone.js) from a class based component to a functional component .
- Replace state with hooks.

We are looking for your review before to move forward next steps.

### CODE REVIEW

File: src/components/resultIcone.js [hunk 1]
```diff
@@ -1,10 +1,3 @@
-import React from 'react';
-export class ResultIcone extends React.Component {
-    constructor(props){
-        super(props)
-        this.props = props;
-    }
-    render(){
-        return <div className = "calculator-result"> <span> {this.props.total}</span> </div>
-    }
+export const ResultIcone = (props) => {
```
Comment (D4nielJ) [inline src/components/resultIcone.js:1]: [OPTIONAL] When trying to use the modulo with 0, it throws an error. It should behave just the same it does the division by 0. You could try to solve this issue so the Calculator behaves as expected in every case.
image

File: src/components/ligneOfButton.js [hunk 2]
```diff
@@ -1,34 +1,25 @@
...
-            <div className ="simple-btn" onClick = {this.props.handler}><span>{btnsText[1]}</span></div>
-            <div className ="op-btn" onClick = {this.props.handler}><span>{btnsText[2]}</span></div>
-          </div>
-          );
-      }
-      return content;
-    }
-  }
\ No newline at end of file
+export const LigneOfButton = (props) => {
```
Comment (D4nielJ) [inline src/components/ligneOfButton.js:1]: - [ ] It's a best practice to deconstruct objects on JS e.g. `{ textBtns, order, handler }`. This avoids a lot of repetition. [If you want to read more about JS best practices, follow this link.](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md)

File: src/components/Calculator.js [hunk 3]
```diff
@@ -1,17 +1,36 @@
-import React from 'react';
```
Comment (D4nielJ) [inline src/components/Calculator.js:1]: [OPTIONAL] On large screens, the calculator appears way down the screen. I don't think this should be the expected behavior. The calculator could probably be better being static around the middle of the window. You might want to give a look at this.

image

**Review by D4nielJ — CHANGES_REQUESTED** [review]
## Hello @Taher-web-dev and @taherhagui, 

You're doing great! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

## Happy coding and stay awesome! 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can always find me on slack!

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## Thank you for implementing all the project requirements, you've done a great job :trophy: There is nothing else to say other than... it's time to merge it :champagne:  :tada: Congratulations

Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**